Refactor Movie and Series Details and for improved scrolling and UI

skaik-mo opened the pull request:
Key changes:
- Replaced `Column` with `verticalScroll` with a `LazyColumn` in `MovieDetailsContent` for better performance with large content.
- Implemented a sticky header using `LazyColumn.stickyHeader` to keep the main movie/series details visible while scrolling.
- Introduced `isScrollingUp` state to dynamically adjust UI elements based on scroll direction, particularly for the `MainMovieOrSeriesDetailsAnimatedContent`.
- Modified `MainMovieOrSeriesDetailsAnimatedContent` to use `EaseIn` and `EaseOut` for smoother fade animations.
- Adjusted the layout of `MinimizedInfoRow` to use `Row` with weights for better alignment.
- Updated the `ReviewCard` to accept `Int` for `rate` instead of `Double`.
- Ensured consistent padding and background colors across components.

skaik-mo committed:
Refactor MovieDetailsScreen for improved scrolling and UI

This commit refactors the `MovieDetailsScreen` to enhance the user experience by implementing a `LazyColumn` for smoother scrolling and a sticky header for the main movie details.

Key changes:
- Replaced `Column` with `verticalScroll` with a `LazyColumn` in `MovieDetailsContent` for better performance with large content.
- Implemented a sticky header using `LazyColumn.stickyHeader` to keep the main movie/series details visible while scrolling.
- Introduced `isScrollingUp` state to dynamically adjust UI elements based on scroll direction, particularly for the `MainMovieOrSeriesDetailsAnimatedContent`.
- Modified `MainMovieOrSeriesDetailsAnimatedContent` to use `EaseIn` and `EaseOut` for smoother fade animations.
- Adjusted the layout of `MinimizedInfoRow` to use `Row` with weights for better alignment.
- Updated the `ReviewCard` to accept `Int` for `rate` instead of `Double`.
- Ensured consistent padding and background colors across components.

skaik-mo committed:
Merge branch 'develop' into fix/scrolling-in-screen-details

* develop: (28 commits)
  change the preview movieItem name to VODItem
  Support RTL in list title section
  adjust back icon direction based on layout direction
  change the movieItem name to VODItem
  get language in the class.
  Fix: Clears focus when login button is clicked in LoginForm
  fix movie card content scale
  fix home items spacing
  fix carousel animation
  edit movie item
  add a language interceptor to automatically include the current language parameter when fetching data in each request.
  refactor: Simplify DefaultTextField by replacing TextFieldValue with String for value management
  refactor: Simplify DefaultTextField by replacing TextFieldValue with String for value management
  fix text field
  add app launcher
  remove unnecessary box
  remove if condition and add animated visibility
  fix recommendation screens crash
  fix month issues in format date
  fix all clickable text's hover effect
  ...

# Conflicts:
#	presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/screen/MovieDetailsScreen.kt

skaik-mo committed:
Refactor MovieDetailsScreen to use navigation callbacks

- Removed `navController` as a direct parameter in `MovieDetailsScreen`.
- Introduced specific navigation callback parameters for actions like:
    - `onClickPlay`
    - `onClickPoster`
    - `navigateToCastDetails`
    - `navigateToMoviesRecommended`
- Updated `MovieDetailsRoute` to pass these new navigation callbacks to `MovieDetailsScreen`.
- Navigation logic within `MovieDetailsScreen` now uses these callbacks instead of directly interacting with `navController`.
- The "Add to Rate" button in the bottom sheet now uses the `onClickPlay` callback.

skaik-mo committed:
Refactor SeriesDetailsContent to use LazyColumn with sticky header

This commit refactors the `SeriesDetailsContent` composable to use a `LazyColumn` for its main content, with the `MainMovieOrSeriesDetailsAnimatedContent` acting as a sticky header.

Key changes:
- Replaced `Column` with `verticalScroll` with a `LazyColumn`.
- Implemented a sticky header for the main series details using `LazyColumn`'s `stickyHeader` feature.
- Updated scroll detection to use `NestedScrollConnection` and `pointerInput` with `detectVerticalDragGestures` to determine if the user is scrolling up.
- The `isScrolled` state for the `MainMovieOrSeriesDetailsAnimatedContent` is now derived from this improved scroll detection.
- Adjusted padding and layout to accommodate the `LazyColumn` structure.
- Updated `duration` in `MainMovieOrSeriesDetailsAnimatedContent` to display the number of seasons.
- Ensured "Show more" for seasons is only visible if there are more than 3 seasons.
- Corrected the type of `rate` in `ReviewCard` from `Int` to `Double`.

Mohamed-Ashraf-2210 commented on presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/screen/MovieDetailsScreen.kt:
```diff
@@ -98,142 +105,148 @@ fun MovieDetailsScreen(
-                onClickPoster = {
-                    navController.navigateToMovieDetails(it.id)
+            stickyHeader {
+                Box(
```
make this box a column and add app bar above AnimatedContent

Mohamed-Ashraf-2210 commented on presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/screen/MovieDetailsScreen.kt:
```diff
@@ -98,142 +105,148 @@ fun MovieDetailsScreen(
+        }
+    }
 
     Column(
```
remove this column

mohannadahmed00 commented on presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/screen/MovieDetailsScreen.kt:
```diff
@@ -98,142 +105,148 @@ fun MovieDetailsScreen(
-                },
-                onClickPoster = {
-                    navController.navigateToMovieDetails(it.id)
+            stickyHeader {
```
the combination of stickyHeader and isScrollingUp logic is causing unexpected scroll behavior.

![20250729-1329-24 4645147](https://github.com/user-attachments/assets/b031a66f-7939-4c76-9de1-47968fac2e5e)

skaik-mo committed:
Refactor MovieDetailsScreen for improved scrolling and layout

- Removed `pointerInput` and `detectVerticalDragGestures` as scrolling detection is now handled by `NestedScrollConnection` and `LazyListState.firstVisibleItemScrollOffset`.
- Encapsulated the `AppBar` and `MainMovieOrSeriesDetailsAnimatedContent` within a `Column` inside the `stickyHeader` for better layout management during scrolling.
- Set `durationAnimation` to 0 in `MainMovieOrSeriesDetailsAnimatedContent` to disable animation during scroll state changes, potentially improving performance.
- Adjusted padding for the `AppBar` within the sticky header.
- The `isScrollingUp` state is now `rememberSaveable` to persist across configuration changes.

skaik-mo committed:
Merge branch 'fix/312-details-screen-scroll-behavior' into fix/scrolling-in-screen-details

* fix/312-details-screen-scroll-behavior:
  Refactor SeriesDetailsContent to use LazyColumn with sticky header

skaik-mo committed:
Refactor SeriesDetailsContent to improve sticky header behavior

This commit refactors the `SeriesDetailsContent` composable to enhance the behavior of the sticky header.

Key changes:
- Moved the `AppBar` inside the `stickyHeader` block within the `LazyColumn`.
- Replaced `remember` with `rememberSaveable` for the `isScrollingUp` state to preserve it across configuration changes.
- Removed the `pointerInput` modifier for drag gesture detection as it's no longer necessary.
- Adjusted padding for the `AppBar`.
- The `LazyColumn` now directly handles the background and system bars padding.

skaik-mo commented:
Preview:
https://github.com/user-attachments/assets/3ec724ec-2321-4f41-a089-d64acdf8dce9

Mohamed-Ashraf-2210 commented on presentation/details/src/main/java/com/giraffe/details/screens/seriesdetails/screen/SeriesDetailsContent.kt:
```diff
@@ -39,50 +52,68 @@ fun SeriesDetailsContent(
+        stickyHeader {
+            Column(
+                modifier = modifier
+                    .background(Theme.color.background.screen)
```
unnecessary background

skaik-mo replied:
We need it because the app bar and header have padding along with the background.

poula22 reviewed (approved):
Great job👏
